packages/indy-sdk/src/dids/IndySdkIndyDidRegistrar.ts

```diff
@@ -0,0 +1,315 @@
+import type { IndyEndpointAttrib } from './didSovUtil'
+import type { IndySdkPool } from '../ledger'
+import type { IndySdk } from '../types'
+import type {
+  AgentContext,
+  Buffer,
+  DidCreateOptions,
+  DidCreateResult,
+  DidDeactivateResult,
+  DidRegistrar,
+  DidUpdateResult,
+} from '@aries-framework/core'
+import type { NymRole } from 'indy-sdk'
+
+import { DidDocumentRole, DidRecord, DidRepository, KeyType, Key } from '@aries-framework/core'
+
+import { IndySdkError } from '../error'
+import { isIndyError } from '../error/indyError'
+import { IndySdkPoolService } from '../ledger'
+import { IndySdkSymbol } from '../types'
+import { assertIndySdkWallet } from '../utils/assertIndySdkWallet'
+import { isLegacySelfCertifiedDid, legacyIndyDidFromPublicKeyBase58 } from '../utils/did'
+
+import { createKeyAgreementKey, indyDidDocumentFromDid, parseIndyDid, verificationKeyForIndyDid } from './didIndyUtil'
+import { addServicesFromEndpointsAttrib } from './didSovUtil'
+
+export class IndySdkIndyDidRegistrar implements DidRegistrar {
+  public readonly supportedMethods = ['indy']
+
+  public async create(agentContext: AgentContext, options: IndySdkIndyDidCreateOptions): Promise<DidCreateResult> {
+    const indySdkPoolService = agentContext.dependencyManager.resolve(IndySdkPoolService)
+    const didRepository = agentContext.dependencyManager.resolve(DidRepository)
+
+    const { alias, role, submitterDid, endpoints } = options.options
+    let did = options.did
+    let didIdentifier: string
+    let verificationKey: Key
+    const privateKey = options.secret?.privateKey
+
+    if (did && privateKey) {
+      return {
+        didDocumentMetadata: {},
+        didRegistrationMetadata: {},
+        didState: {
+          state: 'failed',
+          reason: `Only one of 'privateKey' or 'did' must be provided`,
+        },
+      }
+    }
+
+    try {
+      assertIndySdkWallet(agentContext.wallet)
+
+      // Parse submitterDid and extract namespace based on the submitter did
+      const { namespace: submitterNamespace, id: submitterDidIdentifier } = parseIndyDid(submitterDid)
+      const submitterSigningKey = await verificationKeyForIndyDid(agentContext, submitterDid)
+
+      // Only supports version 1 did identifier (which is same as did:sov)
```

Comment:
Not sure what you mean. I mean the `1` from here: https://hyperledger.github.io/indy-did-method/#nym-transaction-version

Reply:
Yes, I'm referring to that too! When I've read that before implementing IndyVdr did:indy classes, I understood that it would be possible for an app to indicate whether to use 0, 1 or 2 when creating a DID, so I imagined having an extra parameter in the DidRegistrar that would allow us to specify the `transactionVersion`.

So, in case of IndyVdr, we could set `transactionVersion=1` if we wish to register a did:sov and `transactionVersion=2` if we want to register a brand-new did:indy. Even a `transactionVersion=0` would be possible (for super legacy ledgers I think... not sure why it was added).

Do you think this transaction version will be always hard-coded on the indy library? (i.e. 1 for indy-sdk and probably 2 for indy-vdr with did:indy support?)

Reply:
I understand. Yeah we should probably make that configurable and default to `transactionVersion=1`. However we can't set the transaction version I think with the indy-sdk?

Reply:
Yeah, for indy-sdk will be always 1. We just need to make clear about this limitation either in docs or by adding this transactionVersion to DidCreateOptions whose only possible value is '1'. This is of course not for this PR.